subsys/bluetooth/shell/iso.c

```diff
@@ -104,6 +106,35 @@ static int cmd_listen(const struct shell *shell, size_t argc, char *argv[])
 	return err;
 }
 
+static int cmd_peripheral(const struct shell *shell_ctx, size_t argc, char *argv[])
+{
+	struct bt_conn *conns[1];
+	struct bt_iso_chan *chans[1];
+
+	if (!default_conn) {
+		shell_error(shell_ctx, "Not connected");
+		return 0;
+	}
+
+	conns[0] = default_conn;
+	chans[0] = &iso_chan;
+
+	if (!strcmp("tx", argv[1])) {
+		chans[0]->qos->tx = &iso_tx_qos;
+		chans[0]->qos->rx = NULL;
+	} else if (!strcmp("rx", argv[1])) {
+		chans[0]->qos->tx = NULL;
+		chans[0]->qos->rx = &iso_rx_qos;
+	} else if (!strcmp("txrx", argv[1])) {
+		chans[0]->qos->tx = &iso_tx_qos;
+		chans[0]->qos->rx = &iso_rx_qos;
+	} else {
+		shell_error(shell_ctx, "Invalid argument - use tx, rx or txrx");
+	}
```

Comment:
Hmm, I actually wonder if it would make sense to merge the peripheral and listen commands? The above check and assign of the connection pointer can't be done in listen, but the rest here seemingly can. Thoughts?

Reply:
I can check if receiving the connection will reset any of the variables we setup as part the bind/peripheral command. But making it part of listen will limit the app to a single CIS. The code is written to be expandable to more connections - even though I do not know if that planned work.